cli/code_scanner.py

```diff
@@ -531,3 +553,45 @@ def _does_severity_match_severity_threshold(severity: str, severity_threshold: s
         return True
 
     return detection_severity_value >= Severity.try_get_value(severity_threshold)
+
+
+def _get_scan_result(cycode_client, scan_async_result: ScanInitializationResponse,
+                     scan_details: ScanDetailsResponse) -> ZippedFileScanResult:
+    scan_result = ZippedFileScanResult(did_detect=False, detections_per_file=[],
+                                       scan_id=scan_async_result.scan_id,
+                                       report_url=scan_details.report_url)
+    if not scan_details.detections_count:
+        return scan_result
+
+    wait_for_detections_creation(cycode_client, scan_async_result.scan_id, scan_details.detections_count)
+    scan_detections = cycode_client.get_scan_detections_with_paging(scan_async_result.scan_id)
+    scan_result.detections_per_file = _map_detections_per_file(scan_detections)
+    scan_result.did_detect = True
+    return scan_result
+
+
+def wait_for_detections_creation(cycode_client, scan_id: str, expected_detections_count: int):
+    end_polling_time = time.time() + DETECTIONS_COUNT_VERIFICATION_TIMEOUT_IN_SECONDS
+    while time.time() < end_polling_time:
+        scan_persisted_detections_count = cycode_client.get_scan_detections_count(scan_id)
+        if scan_persisted_detections_count == expected_detections_count:
+            return
+        time.sleep(DETECTIONS_COUNT_VERIFICATION_WAIT_INTERVAL_IN_SECONDS)
+
+
+def _map_detections_per_file(detections) -> List[DetectionsPerFile]:
+    detections_per_files = {}
+    for detection in detections:
+        try:
+            detection['message'] = detection['correlation_message']
+            file_name = detection['detection_details']['file_name']
+            if detections_per_files.get(file_name) is None:
+                detections_per_files[file_name] = [DetectionSchema().load(detection)]
+            else:
+                detections_per_files[file_name].append(DetectionSchema().load(detection))
+        except Exception as e:
+            logger.debug("Failed to parse detection: %s", str(e))
+            continue
+
+    return [DetectionsPerFile(file_name=file_name, detections=file_detections)
+            for file_name, file_detections in detections_per_files.items()]
```

Comment:
Why not using `groupby`

Reply:
in order to use groupby (as far as I checked) we will need to install new dependency (like pandas).